File: kotlin/java-basic-example-vs-kotlin/src/main/java/org/example/ex09/Exam09.java
package org.example.ex09;

import lombok.Data;

/**
 * @author : Rene
 * @since : 2023/10/22
 */
public class Exam09 {
	public static void main(String[] args) {

	}

	public Exam09(StoreUser storeUser){
		// service logic

		if (storeUser.getRole().equals("MASTER")){

		} else if (storeUser.getRole().equals("ADMIN")) {

		} else if (storeUser.getRole().equals("USER")){

		}



		switch (storeUser.getRole()){
			case "MASTER":
				break;
			case "ADMIN":
				break;
			case "USER":
				break;
			default:
				// 로직
		}

		try {

		} catch (Exception e){
			if (e instanceof NullPointerException){

			}
		}

	}
}


@Data
class StoreUser{
	private String name;
	private String role;
}


// java에서 이와 같이 분기문을 처리하는 방식 -> 코틀린에서는 ?

File: kotlin/kotlin-basic-example/src/main/kotlin/ex09/Exam09.kt
package ex09

/**
 * @author        : Rene
 * @since         : 2023/10/22
 */
fun main(){

    when(""){
        ""-> {

        }

        "MASTER" ->{

        }
    }



    var result = when(""){
        "MASTER" -> {
            "master"
        }
        else -> {
            "default"
        }
    }



    var any : Any = "";

    // instanceof 와 동일
    when (any) {
        is String ->{}
        is Int ->{}
        is Boolean ->{}
    }


    // switch 문과 매우 동일하게 생겼지만 break을 안해도 된다.
    var exception = RuntimeException()
    when(exception) {
        is NullPointerException ->{}
    }


    // 표현식을 담을 수도 있다.
    var number = 10

    when (val n = number % 2){
        0 -> {println(n)}
        else -> { println(n)}
    }

}


